File: tests/Generator/integration/java/News.java
package org.phpsx.test;

import com.fasterxml.jackson.annotation.JsonGetter;
import com.fasterxml.jackson.annotation.JsonSetter;
import java.net.URI;
import java.time.LocalDate;
import java.time.LocalTime;
import java.time.LocalDateTime;
import java.util.HashMap;

/**
 * An general news entry
 */
public class News {
    private Meta config;
    private HashMap<String, String> inlineConfig;
    private String[] tags;
    private Author[] receiver;
    private Object[] resources;
    private byte[] profileImage;
    private boolean read;
    private Object source;
    private Author author;
    private Meta meta;
    private LocalDate sendDate;
    private LocalDateTime readDate;
    private Period expires;
    private double price;
    private int rating;
    private String content;
    private String question;
    private String version;
    private LocalTime coffeeTime;
    private URI profileUri;
    private String captcha;
    private Object payload;
    @JsonSetter("config")
    public void setConfig(Meta config) {
        this.config = config;
    }
    @JsonGetter("config")
    public Meta getConfig() {
        return this.config;
    }
    @JsonSetter("inlineConfig")
    public void setInlineConfig(HashMap<String, String> inlineConfig) {
        this.inlineConfig = inlineConfig;
    }
    @JsonGetter("inlineConfig")
    public HashMap<String, String> getInlineConfig() {
        return this.inlineConfig;
    }
    @JsonSetter("tags")
    public void setTags(String[] tags) {
        this.tags = tags;
    }
    @JsonGetter("tags")
    public String[] getTags() {
        return this.tags;
    }
    @JsonSetter("receiver")
    public void setReceiver(Author[] receiver) {
        this.receiver = receiver;
    }
    @JsonGetter("receiver")
    public Author[] getReceiver() {
        return this.receiver;
    }
    @JsonSetter("resources")
    public void setResources(Object[] resources) {
        this.resources = resources;
    }
    @JsonGetter("resources")
    public Object[] getResources() {
        return this.resources;
    }
    @JsonSetter("profileImage")
    public void setProfileImage(byte[] profileImage) {
        this.profileImage = profileImage;
    }
    @JsonGetter("profileImage")
    public byte[] getProfileImage() {
        return this.profileImage;
    }
    @JsonSetter("read")
    public void setRead(boolean read) {
        this.read = read;
    }
    @JsonGetter("read")
    public boolean getRead() {
        return this.read;
    }
    @JsonSetter("source")
    public void setSource(Object source) {
        this.source = source;
    }
    @JsonGetter("source")
    public Object getSource() {
        return this.source;
    }
    @JsonSetter("author")
    public void setAuthor(Author author) {
        this.author = author;
    }
    @JsonGetter("author")
    public Author getAuthor() {
        return this.author;
    }
    @JsonSetter("meta")
    public void setMeta(Meta meta) {
        this.meta = meta;
    }
    @JsonGetter("meta")
    public Meta getMeta() {
        return this.meta;
    }
    @JsonSetter("sendDate")
    public void setSendDate(LocalDate sendDate) {
        this.sendDate = sendDate;
    }
    @JsonGetter("sendDate")
    public LocalDate getSendDate() {
        return this.sendDate;
    }
    @JsonSetter("readDate")
    public void setReadDate(LocalDateTime readDate) {
        this.readDate = readDate;
    }
    @JsonGetter("readDate")
    public LocalDateTime getReadDate() {
        return this.readDate;
    }
    @JsonSetter("expires")
    public void setExpires(Period expires) {
        this.expires = expires;
    }
    @JsonGetter("expires")
    public Period getExpires() {
        return this.expires;
    }
    @JsonSetter("price")
    public void setPrice(double price) {
        this.price = price;
    }
    @JsonGetter("price")
    public double getPrice() {
        return this.price;
    }
    @JsonSetter("rating")
    public void setRating(int rating) {
        this.rating = rating;
    }
    @JsonGetter("rating")
    public int getRating() {
        return this.rating;
    }
    @JsonSetter("content")
    public void setContent(String content) {
        this.content = content;
    }
    @JsonGetter("content")
    public String getContent() {
        return this.content;
    }
    @JsonSetter("question")
    public void setQuestion(String question) {
        this.question = question;
    }
    @JsonGetter("question")
    public String getQuestion() {
        return this.question;
    }
    @JsonSetter("version")
    public void setVersion(String version) {
        this.version = version;
    }
    @JsonGetter("version")
    public String getVersion() {
        return this.version;
    }
    @JsonSetter("coffeeTime")
    public void setCoffeeTime(LocalTime coffeeTime) {
        this.coffeeTime = coffeeTime;
    }
    @JsonGetter("coffeeTime")
    public LocalTime getCoffeeTime() {
        return this.coffeeTime;
    }
    @JsonSetter("profileUri")
    public void setProfileUri(URI profileUri) {
        this.profileUri = profileUri;
    }
    @JsonGetter("profileUri")
    public URI getProfileUri() {
        return this.profileUri;
    }
    @JsonSetter("g-recaptcha-response")
    public void setCaptcha(String captcha) {
        this.captcha = captcha;
    }
    @JsonGetter("g-recaptcha-response")
    public String getCaptcha() {
        return this.captcha;
    }
    @JsonSetter("payload")
    public void setPayload(Object payload) {
        this.payload = payload;
    }
    @JsonGetter("payload")
    public Object getPayload() {
        return this.payload;
    }
}

File: tests/Generator/integration/php/News.php
<?php
use PSX\Schema\Attribute\Description;
use PSX\Schema\Attribute\Enum;
use PSX\Schema\Attribute\Key;
use PSX\Schema\Attribute\MaxItems;
use PSX\Schema\Attribute\MaxLength;
use PSX\Schema\Attribute\Maximum;
use PSX\Schema\Attribute\MinItems;
use PSX\Schema\Attribute\MinLength;
use PSX\Schema\Attribute\Minimum;
use PSX\Schema\Attribute\Required;

#[Description('An general news entry')]
#[Required(array('receiver', 'price', 'content'))]
class News implements \JsonSerializable, \PSX\Record\RecordableInterface
{
    protected ?Meta $config = null;
    /**
     * @var \PSX\Record\Record<string>|null
     */
    protected ?\PSX\Record\Record $inlineConfig = null;
    /**
     * @var array<string>|null
     */
    #[MinItems(1)]
    #[MaxItems(6)]
    protected ?array $tags = null;
    /**
     * @var array<Author>|null
     */
    #[MinItems(1)]
    protected ?array $receiver = null;
    /**
     * @var array<Location|Web>|null
     */
    protected ?array $resources = null;
    /**
     * @var resource|null
     */
    protected $profileImage = null;
    protected ?bool $read = null;
    protected Author|Web|null $source = null;
    protected ?Author $author = null;
    protected ?Meta $meta = null;
    protected ?\PSX\DateTime\LocalDate $sendDate = null;
    protected ?\PSX\DateTime\LocalDateTime $readDate = null;
    protected ?\PSX\DateTime\Period $expires = null;
    #[Minimum(1)]
    #[Maximum(100)]
    protected ?float $price = null;
    #[Minimum(1)]
    #[Maximum(5)]
    protected ?int $rating = null;
    #[Description('Contains the main content of the news entry')]
    #[MinLength(3)]
    #[MaxLength(512)]
    protected ?string $content = null;
    #[Enum(array('foo', 'bar'))]
    protected ?string $question = null;
    protected ?string $version = 'http://foo.bar';
    protected ?\PSX\DateTime\LocalTime $coffeeTime = null;
    protected ?\PSX\Uri\Uri $profileUri = null;
    #[Key('g-recaptcha-response')]
    protected ?string $captcha = null;
    protected mixed $payload = null;
    public function setConfig(?Meta $config) : void
    {
        $this->config = $config;
    }
    public function getConfig() : ?Meta
    {
        return $this->config;
    }
    public function setInlineConfig(?\PSX\Record\Record $inlineConfig) : void
    {
        $this->inlineConfig = $inlineConfig;
    }
    public function getInlineConfig() : ?\PSX\Record\Record
    {
        return $this->inlineConfig;
    }
    /**
     * @param array<string>|null $tags
     */
    public function setTags(?array $tags) : void
    {
        $this->tags = $tags;
    }
    public function getTags() : ?array
    {
        return $this->tags;
    }
    /**
     * @param array<Author>|null $receiver
     */
    public function setReceiver(?array $receiver) : void
    {
        $this->receiver = $receiver;
    }
    public function getReceiver() : ?array
    {
        return $this->receiver;
    }
    /**
     * @param array<Location|Web>|null $resources
     */
    public function setResources(?array $resources) : void
    {
        $this->resources = $resources;
    }
    public function getResources() : ?array
    {
        return $this->resources;
    }
    public function setProfileImage($profileImage) : void
    {
        $this->profileImage = $profileImage;
    }
    public function getProfileImage()
    {
        return $this->profileImage;
    }
    public function setRead(?bool $read) : void
    {
        $this->read = $read;
    }
    public function getRead() : ?bool
    {
        return $this->read;
    }
    public function setSource(Author|Web|null $source) : void
    {
        $this->source = $source;
    }
    public function getSource() : Author|Web|null
    {
        return $this->source;
    }
    public function setAuthor(?Author $author) : void
    {
        $this->author = $author;
    }
    public function getAuthor() : ?Author
    {
        return $this->author;
    }
    public function setMeta(?Meta $meta) : void
    {
        $this->meta = $meta;
    }
    public function getMeta() : ?Meta
    {
        return $this->meta;
    }
    public function setSendDate(?\PSX\DateTime\LocalDate $sendDate) : void
    {
        $this->sendDate = $sendDate;
    }
    public function getSendDate() : ?\PSX\DateTime\LocalDate
    {
        return $this->sendDate;
    }
    public function setReadDate(?\PSX\DateTime\LocalDateTime $readDate) : void
    {
        $this->readDate = $readDate;
    }
    public function getReadDate() : ?\PSX\DateTime\LocalDateTime
    {
        return $this->readDate;
    }
    public function setExpires(?\PSX\DateTime\Period $expires) : void
    {
        $this->expires = $expires;
    }
    public function getExpires() : ?\PSX\DateTime\Period
    {
        return $this->expires;
    }
    public function setPrice(?float $price) : void
    {
        $this->price = $price;
    }
    public function getPrice() : ?float
    {
        return $this->price;
    }
    public function setRating(?int $rating) : void
    {
        $this->rating = $rating;
    }
    public function getRating() : ?int
    {
        return $this->rating;
    }
    public function setContent(?string $content) : void
    {
        $this->content = $content;
    }
    public function getContent() : ?string
    {
        return $this->content;
    }
    public function setQuestion(?string $question) : void
    {
        $this->question = $question;
    }
    public function getQuestion() : ?string
    {
        return $this->question;
    }
    public function setVersion(?string $version) : void
    {
        $this->version = $version;
    }
    public function getVersion() : ?string
    {
        return $this->version;
    }
    public function setCoffeeTime(?\PSX\DateTime\LocalTime $coffeeTime) : void
    {
        $this->coffeeTime = $coffeeTime;
    }
    public function getCoffeeTime() : ?\PSX\DateTime\LocalTime
    {
        return $this->coffeeTime;
    }
    public function setProfileUri(?\PSX\Uri\Uri $profileUri) : void
    {
        $this->profileUri = $profileUri;
    }
    public function getProfileUri() : ?\PSX\Uri\Uri
    {
        return $this->profileUri;
    }
    public function setCaptcha(?string $captcha) : void
    {
        $this->captcha = $captcha;
    }
    public function getCaptcha() : ?string
    {
        return $this->captcha;
    }
    public function setPayload(mixed $payload) : void
    {
        $this->payload = $payload;
    }
    public function getPayload() : mixed
    {
        return $this->payload;
    }
    public function toRecord() : \PSX\Record\RecordInterface
    {
        /** @var \PSX\Record\Record<mixed> $record */
        $record = new \PSX\Record\Record();
        $record->put('config', $this->config);
        $record->put('inlineConfig', $this->inlineConfig);
        $record->put('tags', $this->tags);
        $record->put('receiver', $this->receiver);
        $record->put('resources', $this->resources);
        $record->put('profileImage', $this->profileImage);
        $record->put('read', $this->read);
        $record->put('source', $this->source);
        $record->put('author', $this->author);
        $record->put('meta', $this->meta);
        $record->put('sendDate', $this->sendDate);
        $record->put('readDate', $this->readDate);
        $record->put('expires', $this->expires);
        $record->put('price', $this->price);
        $record->put('rating', $this->rating);
        $record->put('content', $this->content);
        $record->put('question', $this->question);
        $record->put('version', $this->version);
        $record->put('coffeeTime', $this->coffeeTime);
        $record->put('profileUri', $this->profileUri);
        $record->put('g-recaptcha-response', $this->captcha);
        $record->put('payload', $this->payload);
        return $record;
    }
    public function jsonSerialize() : object
    {
        return (object) $this->toRecord()->getAll();
    }
}
